arwen/host/execution.go

```diff
@@ -686,6 +698,10 @@ func (host *vmHost) callInitFunction() error {
 		err = host.handleBreakpointIfAny(err)
 	}
 
+	if err == nil {
```

Comment:
return directly?

Reply:
It's not work to return directly, as if err != nil in this point it should be return this err and if you return directly checkFinalGasAfterExit it could be return nil

Reply:
I also propose this refactor for L696-705 :

	_, err := init()
	if err != nil {
		err = host.handleBreakpointIfAny(err)
                if err != nil {
                       return err
                }
	}

        return host.checkFinalGasAfterExit()

Reply:
I would leave like this. It is the correct way to do it.